ros2_ws/Dockerfile

```diff
@@ -1,10 +1,156 @@
+# enable colorized output from ros logging
+RUN echo "export RCUTILS_COLORIZED_OUTPUT=1" >> ${HOME}/.bashrc
+
+# create the credentials to be able to pull private repos using ssh
+USER root
+RUN mkdir /root/.ssh/ && ssh-keyscan github.com | tee -a /root/.ssh/known_hosts
+RUN echo "session required pam_limits.so" | tee --append /etc/pam.d/common-session > /dev/null
+
+FROM base-workspace as ros2-control-iron
+
+RUN apt-get update && apt-get install --no-install-recommends -y \
+  doxygen \
+  librange-v3-dev \
+  python3-jinja2 \
+  python3-typeguard \
+  && rm -rf /var/lib/apt/lists/*
+
+# build as ROS user
+USER ${USER}
+WORKDIR ${ROS2_WORKSPACE}/src
+
+RUN git clone -b 3.18.0 --depth 1 https://github.com/ros-controls/ros2_control.git
+RUN git clone -b 3.14.0 --depth 1 https://github.com/ros-controls/ros2_controllers.git
+# get additional interface dependencies manually
+RUN git clone -b 5.0.0 --depth 1 https://github.com/ros-controls/control_msgs.git
+RUN git clone -b 1.0.2 --depth 1 https://github.com/pal-robotics/backward_ros.git
+RUN git clone -b 3.1.2 --depth 1 https://github.com/ros/diagnostics.git
+RUN git clone -b 2.5.0 --depth 1 https://github.com/ros-controls/realtime_tools.git
+RUN git clone -b 1.7.0 --depth 1 https://github.com/ros2/rcl_interfaces.git
+RUN git clone -b 0.11.0 --depth 1 https://github.com/ros2/test_interface_files.git
+RUN git clone -b 1.16.0 --depth 1 https://github.com/ros/angles.git
+RUN git clone -b ros2 --depth 1 https://github.com/ros-drivers/ackermann_msgs.git
+RUN git clone -b 0.3.3 --depth 1 https://github.com/PickNikRobotics/generate_parameter_library.git
+RUN git clone -b 0.2.2 --depth 1 https://github.com/PickNikRobotics/RSL.git
+RUN git clone -b 1.0.2 --depth 1 https://github.com/PickNikRobotics/cpp_polyfills.git
+RUN git clone -b 3.1.0 --depth 1 https://github.com/ros-controls/control_toolbox.git
+RUN git clone -b 0.1.0 --depth 1 https://github.com/ros-controls/kinematics_interface.git
+
+WORKDIR ${ROS2_WORKSPACE}
+RUN /bin/bash -c "source /opt/ros/${ROS_DISTRO}/setup.bash; colcon build"
+RUN rm -rf ./src
+
+ARG ROS_DISTRO=iron
+FROM ros2-control-${ROS_DISTRO} as ros2-control
+
+FROM environment-variables as final
+COPY --from=ros2-control / /
+
+# start as ROS user
+USER ${USER}
+WORKDIR ${ROS2_WORKSPACE}
+
+# Metadata
+ARG BASE_IMAGE=docker.io/library/ros
 ARG BASE_TAG=iron
+ARG VERSION=v0.0.0
 LABEL org.opencontainers.image.title="AICA ROS 2 image"
 LABEL org.opencontainers.image.description="AICA base ROS 2 image (includes ros2_control)"
 LABEL org.opencontainers.image.version="${VERSION}"
-LABEL org.opencontainers.image.base.name="docker.io/library/ros:${BASE_TAG}"
-LABEL tech.aica.image.metadata='{"type":"base/ws","base":{"name":"docker.io/library/ros","version":"'${BASE_TAG}'"}}'
\ No newline at end of file
+LABEL org.opencontainers.image.base.name="${BASE_IMAGE}:${BASE_TAG}"
+LABEL tech.aica.image.metadata='{"type":"base/ws","base":{"name":"${BASE_IMAGE}:${BASE_TAG}","version":"${VERSION}"}}'
```

Comment:
While the value is correct, the formatting isn't. The previous version used `"'${VAR}'"` for a specific reason (`'` strings do not extrapolate variables). This breaks the metadata, see:
```
                "tech.aica.image.metadata": "{\"type\":\"base/ws\",\"base\":{\"name\":\"${BASE_IMAGE}:${BASE_TAG}\",\"version\":\"${VERSION}\"}}"
```